C#: Fix external API name for nested types

koesie10 opened the pull request:
This fixes the name of reported external APIs for nested types. The `getDeclaringType().getUnboundDeclaration()`'s `toString()` method reports the name of the type, but not the name of the declaring type. This results in missing information in the `UnsupportedExternalAPIs.ql` query.

For example, previously it would report:

```
GitHub.Nested#NestedClass.Test()
```

However, the `NestedClass` class does not exist in the namespace and is only a nested type within `MyFirstClass`. The correct name should be:

```
GitHub.Nested#MyFirstClass+NestedClass.Test()
```

This name also matches the format of MaD.

hvitved reviewed (commented):
LGTM. Could we please have a test case for this added to `ql/test/query-tests/Telemetry/LibraryUsage/ExternalLibraryUsage.cs`?

koesie10 commented:
> Could we please have a test case for this added to `ql/test/query-tests/Telemetry/LibraryUsage/ExternalLibraryUsage.cs`?

Thanks for the suggestion and pointer! I've added a call to `System.Collections.Generic#Dictionary<,>+KeyCollection.GetEnumerator()` in that file and updated the `.expected` files.
